gridcapa-swe-commons/src/main/java/com/farao_community/farao/gridcapa_swe_commons/shift/SweNetworkShifter.java

```diff
@@ -163,21 +132,69 @@ public void shiftNetwork(double stepValue, Network network) throws GlskLimitatio
+            String zoneId = entry.getKey();
+            double asked = entry.getValue();
+            String logApplyingVariationOnZone = String.format("[%s] : Applying variation on zone %s (target: %.2f)", direction, zoneId, asked);
+            LOGGER.info(logApplyingVariationOnZone);
+            double done = zonalScalable.getData(zoneId).scale(network, asked, scalingParameters);
+            if (Math.abs(done - asked) > DEFAULT_SHIFT_EPSILON) {
+                String logWarnIncompleteVariation = String.format("[%s] : Incomplete variation on zone %s (target: %.2f, done: %.2f)",
+                        direction, zoneId, asked, done);
+                LOGGER.warn(logWarnIncompleteVariation);
+                limitingCountries.add(zoneId);
+            }
+        }
+        // During the shift some generators linked to the main network with a transformers are not connected correctly
+        // Waiting for a fix in powsybl-core, we connect the transformers linked to these generators to be correctly connected to the main network component
+        scalableGeneratorConnector.connectGeneratorsTransformers(network, PRE_PROCESSING_COUNTRIES);
+
+        if (!limitingCountries.isEmpty()) {
+            StringJoiner sj = new StringJoiner(", ", "There are Glsk limitation(s) in ", ".");
+            limitingCountries.forEach(sj::add);
+            LOGGER.error("[{}] : {}", direction, sj);
+            throw new GlskLimitationException(sj.toString());
         }
     }
 
-    private void preProcessNetwork(Network network, ScalableGeneratorConnector scalableGeneratorConnector, String initialVariantId, String processedVariantId, String workingVariantCopyId) throws ShiftingException {
+    private void preProcessNetwork(Network network, ScalableGeneratorConnector scalableGeneratorConnector, GeneratorLimitsHandler generatorLimitsHandler, String initialVariantId, String processedVariantId, String workingVariantCopyId) throws ShiftingException {
         network.getVariantManager().cloneVariant(initialVariantId, processedVariantId, true);
         network.getVariantManager().setWorkingVariant(processedVariantId);
-        scalableGeneratorConnector.prepareForScaling(network, CONNECT_GENERATORS_IN_COUNTRIES);
+        scalableGeneratorConnector.fillGeneratorsInitialState(network, PRE_PROCESSING_COUNTRIES);
+        // here set working variant generators pmin and pmax values to default values
+        // so that glsk generator pmin and pmax values are used
+        generatorLimitsHandler.setPminPmaxToDefaultValue(network, PRE_PROCESSING_COUNTRIES);
         network.getVariantManager().cloneVariant(processedVariantId, workingVariantCopyId, true);
         network.getVariantManager().setWorkingVariant(workingVariantCopyId);
     }
 
+    private static ScalingParameters getScalingParameters() {
+        ScalingParameters scalingParameters = new ScalingParameters();
+        scalingParameters.setIterative(true);
```

Comment:
It looks like setIterative is deprecated since 6.0.0, you might want to use 
```suggestion
        scalingParameters.setPriority(ScalingParameters.DEFAULT_PRIORITY);
```

Reply:
I think setIterative(true) is equivalent to ScalingParameters.Priority.RESPECT_OF_VOLUME_ASKED